File: for-doc-java/src/main/java/org/doc/j/model/Author.java
package org.doc.j.model;

import org.babyfish.jimmer.sql.*;

import java.util.List;

@Entity
public interface Author {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    String firstName();

    String lastName();

    Gender gender();

    @ManyToMany(mappedBy = "authors")
    List<Book> books();
}

File: for-doc-kotlin/src/main/kotlin/org/doc/k/model/Author.kt
package org.doc.k.model

import org.babyfish.jimmer.sql.*

@Entity
interface Author {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    val firstName: String

    val lastName: String

    val gender: Gender

    @ManyToMany(mappedBy = "authors")
    val books: List<Book>
}